pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
----- Original Message -----
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace",
> >  		event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be
> > normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind ==
> > "Node" {
> 
> Thanks, I just had a chat w/ erictune about this. I may make some cleanup PRs
> to try and make some of that code manageable again, but that's another
> story.
> 
> Do you think we should allow events as both namespaced and not namespaced?
> otherwise where should we put events for root resources?

I think for right now we should allow events to be both - namespace = "" means no namespace (NamespaceNone) and if you specify that for a namespaced resource the event was just improperly formed?

> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4396/files#r24941447